app/src/main/java/com/nutomic/syncthingandroid/activities/FolderActivity.java

```diff
@@ -423,13 +432,13 @@ private void initFolder() {
                 : generateRandomFolderId();
         mFolder.label = getIntent().getStringExtra(EXTRA_FOLDER_LABEL);
         if (Build.VERSION.SDK_INT != Build.VERSION_CODES.M) {
-            // Scan every 3 days (in case inotify dropped some changes)
-            mFolder.rescanIntervalS = 259200;
+            // Scan every hour (in case real time change detection failed)
+            mFolder.rescanIntervalS = 3600;
         }
         else {
             // FileObserver is broken on Marshmallow.
             // https://github.com/syncthing/syncthing-android/issues/787
-            mFolder.rescanIntervalS = 60;
+            mFolder.rescanIntervalS = 300;
```

Comment:
I found this consumed too much battery during testing because if you have very large folders syncing to a mobile device the scan will take half of the scan interval time and the rescan will soon occur. I would advise, as our app users cannot easily tweak the rescan interval, giving them a more "lightweight" interval. With the file watcher coming up soon, users that need instant-syncing of files can (and will get) the watcher turned on to fit their needs. This is more a strategy, consider a user turning off instant-sync in the UI it means "the user has time to wait for syncing".

Reply:
Isn't that marshmallow problem FolderObserver specific? Such that now with the FolderObserver removed, this condition is no longer needed and the rescan interval can be set the same for all android versions.

Reply:
This is fallback scans isn't it?
I thought the watcher adjusts the interval automatically?

Reply:
No fallback, that idea was discarded early. Full scan interval is completely independent from fs watcher. Just in the edit folder dialog we propose default values depending on whether fs watcher is enabled or not.

Reply:
I made a new PR where we can decide how we want the interval. So far, I followed syncthing core where it is 3600s. https://github.com/syncthing/syncthing-android/pull/1069